hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/ddl/TestSpark3DDL.scala

```diff
@@ -779,7 +782,9 @@ class TestSpark3DDL extends HoodieSparkSqlTestBase {
           spark.read.format("hudi").load(tablePath).show(false)
 
           val updatesAgain = QuickstartUtils.convertToStringList(dataGen.generateUpdates(10))
-          val dfAgain = spark.read.json(spark.sparkContext.parallelize(updatesAgain, 2)).withColumn("fare", expr("cast(fare as string)"))
+          val dfAgain = spark.read.json(spark.sparkContext.parallelize(updatesAgain, 2)).
+            withColumn("fare", expr("cast(fare as string)")).
+            withColumn("ts", lit("20240404000015")) // to make test determinate for HOODIE_AVRO_DEFAULT payload
           dfAgain.write.format("hudi").
```

Comment:
Does the `withColumn` overrides the overall column of the original dataset?

Reply:
i think yes, it sets new value of column ts for all rows in dataframe.